src/Application/Adapters/WordPress/Preset.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\PhpInsights\Application\Adapters\WordPress;
+
+use NunoMaduro\PhpInsights\Domain\Contracts\Preset as PresetContract;
+use PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\ForbiddenFunctionsSniff;
+use SlevomatCodingStandard\Sniffs\TypeHints\TypeHintDeclarationSniff;
+
+/**
+ * @internal
+ */
+final class Preset implements PresetContract
+{
+    /**
+     * {@inheritDoc}
+     */
+    public static function getName(): string
+    {
+        return 'wordpress';
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public static function get(): array
+    {
+        return [
+            'exclude' => [
+                'wp-admin',
+                'wp-includes',
+            ],
+            'remove' => [
+                TypeHintDeclarationSniff::class,
+                \WooCommerce::class,
+            ],
+            'config' => [
+                ForbiddenFunctionsSniff::class => [
+                    'forbiddenFunctions' => [
+                        'var_dump' => null,
+                    ],
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public static function shouldBeApplied(array $composer): bool
+    {
+        /** @var string[] $requirements */
+        $requirements = $composer['require'] ?? [];
+
+        foreach (array_keys($requirements) as $requirement) {
+            $requirement = (string) $requirement;
+
+            if (strpos($requirement, 'johnpbloch/wordpress') !== false) {
```

Comment:
Everyone on wordpress uses composer?

Reply:
what I know about this composer is not usually used with wordpress - so imho you cannot trust this.

Reply:
I would prefer something that could that detect if it's a wordpress project, such as: presence of `wp_admin` folder or something.

Reply:
Certainly, composer is not officially supported, altho this is the most common way of working it through composer, not the only one I might add.

The way of installing this packages is through composer too, so my thoughts was, no one outside composer will install this package to analyse the code on his / her WordPress project.

In my imho this is just fine, and could be the base to extend the discovery of WordPress.

If we think, someone will install / use packages with composer without manipulating WordPress also with composer (ofc it can happen) then yes, I'll suggest to look for wp-config.php instead or wp-settings.php

What could be the best option? would it possible provide two ways of discovery? one with composer packages the other with looking files?

Reply:
Exactly, I think you should try to see if the composer key exists, then, if not, you should try to inspect the folder structure. You may need to change phpinsights code to do this.

Reply:
I could use Finder to check for those files if no composer key or even composer is been use.